networkd/network_server.ml

```diff
@@ -85,12 +87,22 @@ let set_dns_interface _ dbg ~name =
 	debug "Setting DNS interface to %s" name;
 	config := {!config with dns_interface = Some name}
 
+let compare_driver_version vs1 vs2 () =
+	try
+		let version_of vs = List.map int_of_string (String.split '.' vs) in
+		let first_non_zero lst = List.fold_left (fun a b -> if (a<>0) then a else b) 0 lst in
+		(first_non_zero (List.map2 compare (version_of vs1) (version_of vs2)) <= 0)
```

Comment:
I think this function is a bit hard to follow, and it took me a while to figure out how it is meant to work. Also, it is not obvious what the return value of the function actually means. Also, it looks like `version_of` is really more like a `list_of_version`...

How about something like this:
```
(* Returns `true` is vs1 is older than vs2 *)
let is_older_version vs1 vs2=
  let list_of_version vs = List.map int_of_string (String.split '.' vs) in
  let rec loop vs1' vs2' =
    match vs1', vs2' with
    | [], _ | _, [] -> false
    | a :: _, b :: _ when a < b -> true
    | _ :: tl1, _ :: tl2 -> loop tl1 tl2
  in
  loop (list_of_version vs1) (list_of_version vs2)
```